code/model/SiteTree.php

```diff
@@ -2497,6 +2542,37 @@ public function doRestoreToStage() {
 	}
 
 	/**
+	 * Removes the page from both live and stage
+	 *
+	 * @return bool Success
+	 */
+	public function doArchive() {
+		if($this->doUnpublish()) {
+			$this->delete();
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the current user is allowed to archive this page.
+	 * If extended, ensure that both canDelete and canDeleteFromLive are extended also
+	 *
+	 * @param Member $member
+	 * @return bool
+	 */
+	public function canArchive($member = null) {
+		$member = $member ?: Member::currentUser();
+		
+		// Standard mechanism for accepting permission changes from extensions
+		$extended = $this->extendedCan('canArchive', $member);
+		if($extended !==null) return $extended;
+
+		// Check both stages
+		return $this->canDeleteFromLive($member) && $this->canDelete($member);
```

Comment:
Is canDeleteFromLive always needed? If a page is only staged, then a user who has permission to delete from stage but not from live should be able to delete (or archive :p ).